app.py

```diff
@@ -1868,23 +1883,37 @@ def logout():
     logout_user()
     return redirect(url_for('index'))
 
+# #########################################################
+@app.route('/downloadFeedBack/<string:project_id>/<string:evaluation_name>/<string:show_score>', methods=['GET', 'POST'])
+@login_required
+def downloadFeedBack(project_id, evaluation_name, show_score):
+
+    project = Permission.query.filter_by(project_id=project_id).first()         #find project
+
+    evaluation_name_find_in_db = Evaluation.query.filter_by(project_owner=current_user.username, project_name=project.project).all()    #look up all evaluations associated with project
+    
+    now = datetime.now().time()
+    current_time = now.strftime("%H_%M_%S")
+    filename = "feedback_{}_{}.zip".format(evaluation_name, current_time)
+    zipObj = ZipFile(filename,'w')
+
+    at_least_one_match = False
+
+    path_to_evalutaion_name = "{}/{}/{}/evalutaion/{}".format(base_directory, project.owner, project.project,evaluation_name)    #make directory to place each evaluation under a file named after eva_name
+
+    for roots, _, files in os.walk(path_to_evalutaion_name):
+        for file in files:
+            filepath = os.path.join(roots,file)
+            at_least_one_match = True
+            zipObj.write(filepath)
+    zipObj.close()
+
+    if not at_least_one_match:
```

Comment:
If user clicks Download Feedback before sending emails, they get an error screen. I think it would be common for a user to want to see the feedback *before* emails are sent. Let's put the HTML file generation in a separate function and have both downloadFeedback and send emails call that function.

Reply:
Also, make sure to use include the show_score argument when downloadFeedback calls this function